### PR TITLE
Modal: responsive footer

### LINKED ISSUE #485: Modal: responsive footer buttons
https://app.zeplin.io/project/5efe8c616e1cfb13f088fc06/screen/5f1543a7c5b99d9c703e1693

image

### PR DESCRIPTION
**Pull request by rosslo**

### CODE REVIEW

File: src/Display/Modal/Modal.tsx [hunk 1]
```diff
@@ -61,6 +68,25 @@ const Modal = (props: Props) => {
       document.removeEventListener('keydown', escapeKeyEventListener, false);
   }, [isVisible, onClose]);
 
+  React.useLayoutEffect(() => {
```
Comment (ritikamotwani) [inline src/Display/Modal/Modal.tsx]: Should we name this function, as there's already one useLayoutEffect hook called, would be better to know the responsibility for this.

File: src/Display/Modal/Modal.tsx [hunk 2]
```diff
@@ -61,6 +68,25 @@ const Modal = (props: Props) => {
...
+    if (footerElementCount > 1) {
+      const checkIsFooterChildrenInMultiLines = () => {
```
Comment (ritikamotwani) [inline src/Display/Modal/Modal.tsx:72]: @rosslo ,  should this function be outside the hook. It's just weird to me that inside the hook there's a function in the hook, then it's called and then an event listener is set.

**Comment by ritikamotwani** [comment]
Another question here @rosslo . Is there a need to check if the top button is a primary button? 
For example if I keep button 1 as primary and button 2 as ghost, then button 2 would be on the top.
Or maybe we can remind the designers to add this in the guideline that the display of buttons would be like, if there's a primary button then it would be at the second position.

File: src/Display/Modal/Modal.tsx (hunk 1 above)
Comment (rosslo) [reply src/Display/Modal/Modal.tsx]: ok!

File: src/Display/Modal/Modal.tsx (hunk 2 above)
Comment (rosslo) [reply src/Display/Modal/Modal.tsx:72]: If we put this function be outside the hook, we will need to wrap useCallback for it, I think we can just declare it inside useLayoutEffect to save a bit performance.

**Comment by rosslo** [comment]
> Or maybe we can remind the designers to add this in the guideline that the display of buttons would be like, if there's a primary button then it would be at the second position.

Yeah, good suggestion, will remind designers later. 🙏

**Comment by ritikamotwani** [comment]
> Yeah, good suggestion, will remind designers later. 🙏

Cool :P
I think we should open an issue to add the guidelines to storybook after the designers send a proper write up.
1. Only 1 or 2 buttons preferred to be there in the modal.
2. The primary button should be the second one, so that the devs know about it.

**Review by ritikamotwani — APPROVED** [review]

**Comment by ritikamotwani** [comment]
LGTM @rosslo !

**Pull request merged**